environment.go

```diff
@@ -1237,15 +1230,42 @@ func (e *envState) Close(ctx context.Context) error {
 	return catcher.Resolve()
 }
 
-// getClientConfig should be called once at startup and looks at the
-// current environment and loads all currently available client
-// binaries for use by the API server in presenting the settings page.
-//
-// If there are no built clients, this returns an empty config
-// version, but does *not* error.
-func getClientConfig(baseURL, s3BaseURL string) (*ClientConfig, error) {
-	c := &ClientConfig{}
-	c.LatestRevision = ClientVersion
+func (e *envState) populateS3ClientConfig(ctx context.Context, versionID string) error {
+	bucket, err := pail.NewS3Bucket(pail.S3Options{
+		Name:   e.settings.Providers.AWS.BinaryClient.Bucket,
+		Region: DefaultEC2Region,
+		Credentials: pail.CreateAWSCredentials(
+			e.settings.Providers.AWS.BinaryClient.Key,
+			e.settings.Providers.AWS.BinaryClient.Secret,
+			"",
+		),
+	})
+	if err != nil {
+		return errors.Wrap(err, "constructing pail bucket")
+	}
+
+	prefix := fmt.Sprintf("%s/%s", s3ClientsPrefix, versionID)
+	c := &ClientConfig{
+		LatestRevision: ClientVersion,
+		S3URLPrefix: fmt.Sprintf("https://%s.s3.amazonaws.com/%s",
+			e.settings.Providers.AWS.BinaryClient.Bucket,
+			prefix,
+		),
+	}
+	if err = c.populateClientBinaries(ctx, bucket, prefix); err != nil {
+		return errors.Wrap(err, "populating client binaries")
+	}
+
+	e.clientConfig = c
+	return nil
+}
+
+func (e *envState) populateLocalClientConfig() error {
+	c := &ClientConfig{LatestRevision: ClientVersion}
+	if e.settings.HostInit.S3BaseURL != "" {
+		c.S3URLPrefix = fmt.Sprintf("%s/%s", e.settings.HostInit.S3BaseURL, BuildRevision)
```

Comment:
Just for my own education, could you explain this part? I assumed that this function would just be the same as the old code.

Reply:
Also will this function go away after we switch to kanopy? Or do you think App will need to maintain it? In which case it may be helpful to add some comments here about whats' going on

Reply:
The idea of the `S3URLPrefix ` is that after the initial setup they can be treated identically. So no matter whether the S3 binaries are in the `upload-clis` place or the `build-and-push` place we can get a URL by appending the path to a prefix.

This is indeed intended to be temporary. After the switchover it will look more like https://github.com/evergreen-ci/evergreen/pull/7349, where there was a single source for the S3 binaries. I'll try to add some more color with comments, though.